addons/repair/models/repair.py

```diff
@@ -236,13 +236,18 @@ def onchange_partner_id(self):
             self.partner_invoice_id = addresses['invoice']
             self.pricelist_id = self.partner_id.property_product_pricelist.id
 
-    @api.depends('company_id')
+    @api.depends('company_id', 'picking_id', 'lot_id')
     def _compute_location_id(self):
         for order in self:
-            if order.company_id:
-                if order.location_id.company_id != order.company_id:
-                    warehouse = self.env['stock.warehouse'].search([('company_id', '=', order.company_id.id)], limit=1)
-                    order.location_id = warehouse.lot_stock_id
+            if order.lot_id and order.lot_id.quant_ids:
+                quants = order.lot_id.quant_ids.filtered(lambda q: q.quantity > 0 and q.location_id.usage in ['internal', 'customer', 'transit'])
+                if quants.location_id:
+                    order.location_id = quants.location_id[-1]
```

Comment:
in your commit message you say it takes the first location possible but in the code it takes the last. Anyway, there's still an issue when you select a lot with quant in location A then switch to a lot with quant only in Inventory Adjustment. It will keep the location of the first quant instead of putting False. Just do this instead:
```suggestion
                order.location_id = quants.location_id[:1]
```
that way, if there's no possible location, it will be False instead of keeping the current location.

The rest is ok

Reply:
@Sehnde Done